tests/conftest.py

```diff
@@ -52,26 +63,81 @@ def create_client(self):
         command = ['xterm', '-hold', '-class', wmclass, '-e', 'true']
         # enforce a hook when the window appears
         self.call('rule', 'once', 'class='+wmclass, 'hook=here_is_'+wmclass)
-        # run process that wait for the hook
-        hc_wait = subprocess.Popen([self.HC_PATH, '--wait', 'rule', 'here_is_'+wmclass],
-                    env=self.env,
-                    stdout=subprocess.PIPE)
-        # start the process...
-        proc = subprocess.Popen(command,env=self.env)
+        proc = subprocess.Popen(command, env=self.env)
         # once the window appears, the hook is fired, and the --wait exits:
-        hc_wait.wait(2)
-        winid = hc_wait.stdout.read().decode().rstrip('\n').split('\t')[-1]
+        winid = self.wait_for_window_of(wmclass)
 
         return SimpleNamespace(proc=proc, winid=winid)
 
+    def wait_for_window_of(self, wmclass):
+        """Wait for a rule hook of the form "here_is_" + wmclass """
+        # we don't need to investigate the rule here
```

Comment:
I'm not sure what is meant here by "investigate the rule"

Reply:
We don't need to read the second argument of the hook and don't need to check that is indeed equals `"here_is_" + wmclass`. But we need to check this once we create clients simultaneously.